homeassistant/components/matter/climate.py

```diff
@@ -0,0 +1,216 @@
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingAndHeatingWithReheat,
+        ):
+            return [HVACMode.HEAT, HVACMode.COOL]
+        if value in (
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingOnly,
+            clusters.Thermostat.Enums.ThermostatControlSequence.kCoolingWithReheat,
+        ):
+            return [HVACMode.COOL]
+        if value in (
+            clusters.Thermostat.Enums.ThermostatControlSequence.kHeatingOnly,
+            clusters.Thermostat.Enums.ThermostatControlSequence.kHeatingWithReheat,
+        ):
+            return [HVACMode.HEAT]
+        return [HVACMode.OFF]
+
+    @property
+    def hvac_action(self) -> HVACAction | None:
+        """Return the current running hvac operation (if supported)."""
+        if running_state := self.get_matter_attribute_value(
+            clusters.Thermostat.Attributes.ThermostatRunningState
+        ):
+            if running_state == 0:
+                return HVACAction.HEATING
+            if running_state == 1:
+                return HVACAction.COOLING
+            if running_state == 2:
+                return HVACAction.FAN
+        if running_mode := self.get_matter_attribute_value(
+            clusters.Thermostat.Attributes.ThermostatRunningMode
+        ):
+            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kCool:
+                return HVACAction.COOLING
+            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kHeat:
+                return HVACAction.HEATING
+            if running_mode == clusters.Thermostat.Enums.ThermostatRunningMode.kOff:
+                return HVACAction.OFF
+        return None
+
+    def _get_temperature_in_degrees(self, attribute) -> float | None:
+        """Return the scaled temperature value for the given attribute."""
+        if value := self.get_matter_attribute_value(attribute):
+            return value / TEMPERATURE_SCALING_FACTOR
+        return None
+
+    @property
+    def current_temperature(self) -> float | None:
+        """Return the current temperature."""
+        return self._get_temperature_in_degrees(
+            clusters.Thermostat.Attributes.LocalTemperature
+        )
+
+    @property
+    def target_temperature(self) -> float | None:
+        """Return the temperature we try to reach."""
+        match self.hvac_mode:
+            case HVACMode.COOL:
+                return self._get_temperature_in_degrees(
+                    clusters.Thermostat.Attributes.OccupiedCoolingSetpoint
+                )
+            case HVACMode.HEAT:
+                return self._get_temperature_in_degrees(
+                    clusters.Thermostat.Attributes.OccupiedHeatingSetpoint
+                )
+            case HVACMode.AUTO:
+                # When the system mode is set to "auto," there is no target temperature; instead, there is a target temperature low and high.
```

Comment:
Can we set the low and high target temps for this case?

Reply:
Yes, we can